network/write_buffer.go

```diff
@@ -105,8 +105,9 @@ func (w *writeBuffer[T]) Send(ctx context.Context) {
 }
 
 func (w *writeBuffer[T]) send(bb []byte, s signalsInfo, ctx context.Context) {
+	bbLen := len(bb)
```

Comment:
This is a bit of future proofing to handle if we do something with the buffer like reuse, it should ideally be the length when it was called.